src/libaktualizr/primary/CMakeLists.txt

```diff
@@ -26,8 +26,16 @@ if (BUILD_OSTREE)
         ENVIRONMENT LD_PRELOAD=$<TARGET_FILE:ostree_mock>
         LABELS "noptest")
     target_link_libraries(t_aktualizr_fullostree virtual_secondary)
+    add_aktualizr_test(NAME download_nonostree SOURCES  download_nonostree_test.cc PROJECT_WORKING_DIRECTORY ARGS $<TARGET_FILE:aktualizr-repo> ${PROJECT_BINARY_DIR}/ostree_repo)
+    set_target_properties(t_download_nonostree PROPERTIES LINK_FLAGS -Wl,--export-dynamic)
+    add_dependencies(t_download_nonostree ostree_mock aktualizr-repo make_ostree_sysroot)
+    set_tests_properties(test_download_nonostree PROPERTIES
+        ENVIRONMENT LD_PRELOAD=$<TARGET_FILE:ostree_mock>
+        LABELS "noptest")
```

Comment:
I think you can do without the `set_target_properties` and `ostree_mock` references here.